### DISCUSSION TITLE
New ToDo-List Tutorial is alive

### DISCUSSION THREAD

**Discussion by timunie** [AvaloniaUI/Avalonia#15783] [General]
After a huge rewrite, we are happy to announce that we have a new beginners tutorial: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CompleteApps/SimpleToDoList

Please let us know your feedback 😄

**Comment by shimpe** [comment]
Maybe one question: Avalonia ships with reactiveui, yet the first and simplest tutorial about Avalonia starts by replacing reactiveui with communitytoolkit mvvm, without giving a motivation for doing so. So, as a beginner, I wonder: why? Are both systems mutually exclusive, and if so, when is one preferred over the other?

**Reply by ykarpeev** [reply]
I think it's because then the tutorial will be about teaching avalonia + reactiveUI rather than focusing on avalonia exclusively.

**Reply by timunie** [reply]
It depends how you create the project these days. You can have both choices. In fact, Avalonia can work with almost any MVVM framework. 

The reason RxUI was the default and still is for some templates is just history. when Avalonia started, it was the best available. Nowadays also the CommunityToolkit is there and has some advantages if you don't need all functions from RxUI. I prefer this package therefore when applicable but this is more a religion choice 🤓

To make confusion complete, you can also use both packages side by side and I do it in more complex Apps to have the best of both worlds. 

Welcome and happy coding

**Reply by shimpe** [reply]
3 weeks later I'm still a basic (or perhaps early intermediate) user, but I've found out by now that ReactiveUI provides ReactiveUI.SourceGenerators which appears to make it just as convenient to use as CommunityToolkit.MVVS. I think it will be my preferred choice from now on.

**Reply by VisualMelon** [reply]
This is something that is configurable in (some of) the templates (including the MVVM one): would it be best to make it configurable in all of them and be more explicit about how to create the project in the first place? There's obviously value in showing which bits are needed for ReactiveUI vs CommunityToolkit, but I'm with @shimpe in thinking a beginner's tutorial isn't the place: it would just irritate me.

![image](https://github.com/user-attachments/assets/0fb74b15-39a0-4e81-b637-93393a1d59db)
